shared/actions/fs/platform-specific.desktop.js

```diff
@@ -350,6 +349,42 @@ const loadUserFileEdits = (state: TypedState, action) =>
     }
   })
 
+const openFilesFromWidget = (state: TypedState, {payload: {path, type}}: FsGen.OpenFilesFromWidgetPayload) => {
+  let navigation = () => []
+  if (path) {
+    const parentFolder = Types.getPathFromElements(Types.getPathElements(path).slice(0, -1))
+    navigation = () => [
+      Saga.put(
+        navigateTo([
+          Tabs.fsTab,
+          {
+            props: {path: parentFolder},
+            selected: 'folder',
+          },
+        ])
+      ),
+      Saga.put(
+        navigateAppend([
+          {
+            props: {path},
+            selected: type === 'folder' ? 'folder' : 'preview',
```

Comment:
If `type === 'folder'`, do we still want to have its parent path in the route?

Reply:
Yeah I think so.